python/cudf_polars/cudf_polars/experimental/sort.py

```diff
@@ -41,5 +388,116 @@ def _(
             partition_info[new_node] = PartitionInfo(count=1)
         return new_node, partition_info
 
-    # Fallback
-    return _lower_ir_fallback(ir, rec, msg="Sort does not support multiple partitions.")
+    # Check sort keys
+    if not all(
+        expr.is_pointwise for expr in traversal([e.value for e in ir.by])
+    ):  # pragma: no cover
+        return _lower_ir_fallback(
+            ir,
+            rec,
+            msg="sort does not support multiple partitions for non-pointwise `by` keys.",
+        )
+
+    # Extract child partitioning
+    child, partition_info = rec(ir.children[0])
+
+    # Handle single-partition case
+    if partition_info[child].count == 1:
+        single_part_node = ir.reconstruct([child])
+        partition_info[single_part_node] = partition_info[child]
+        return single_part_node, partition_info
+
+    local_sort_node = ir.reconstruct([child])
+    partition_info[local_sort_node] = partition_info[child]
+
+    shuffle = ShuffleSorted(
+        ir.schema,
+        ir.by,
+        ir.order,
+        ir.null_order,
+        rec.state["config_options"],
+        local_sort_node,
+    )
+    partition_info[shuffle] = partition_info[child]
+
+    # We sort again locally.
+    assert ir.zlice is None  # zlice handling would be incorrect without adjustment
+    # TODO: We should integrate this with the shuffling step as a merge or merge_stable.
+    #       (as of 25.06, merge_stable is missing in libcudf). (May need zslice in shuffle step)
+    final_sort_node = ir.reconstruct([shuffle])
+    partition_info[final_sort_node] = partition_info[shuffle]
+
+    return final_sort_node, partition_info
+
+
+@generate_ir_tasks.register(ShuffleSorted)
+def _(
+    ir: ShuffleSorted, partition_info: MutableMapping[IR, PartitionInfo]
+) -> MutableMapping[Any, Any]:
+    # Extract "shuffle_method" configuration
+    assert ir.config_options.executor.name == "streaming", (
+        "'in-memory' executor not supported in 'generate_ir_tasks'"
+    )
+    shuffle_method = ir.config_options.executor.shuffle_method
+
+    by = [ne.value.name for ne in ir.by if isinstance(ne.value, Col)]
+    if len(by) != len(ir.by):
+        # We should not reach here as this is checked in the lower_ir_node
+        raise NotImplementedError("Sorting columns must be column names.")
+
+    sort_boundaries_name, graph = _sort_boundaries_graph(
+        get_key_name(ir.children[0]),
```

Comment:
Nit: Minor preference to do `child, = ir.children` earlier in this function, and then refer to `child` instead of `ir.children[0]` throughout.